config/lib/authorization.js

```diff
@@ -0,0 +1,91 @@
+﻿'use strict';
+
+var _ = require('lodash'),
+  config = require('../config'),
+  mongoose = require('mongoose'),
+  mg = require('./mongoose'),
+  jwt = require('jsonwebtoken'),
+  chalk = require('chalk');
+
+// build token auth service
+var tokenAuth = {
+  signToken: signAuthToken,
+  authorize: authorizeRequest
+};
+
+// export the token auth service
+module.exports = tokenAuth;
+
+// Sign a token
+function signAuthToken (user) {
+  return new Promise(function (resolve, reject) {
+    try {
+      var expiration = Date.now() + (config.tokenAuth.expiresInSeconds * 1000);
+
+      var payload = {
+        user: user._id,
+        expiration: expiration
+      };
+
+      var options = {
+        expiresIn: config.tokenAuth.expiresInSeconds
+      };
+
+      var tokenInfo = {
+        token: jwt.sign(payload, config.tokenAuth.secret, options),
+        expiration: expiration
```

Comment:
Why are you putting the expiration here?  Would it be best to just decode the token and if the expiration date is passed then they have to login again to get a new token?  I can't think of a reason to have a token and expiration here.  I would just resolve the token.